_posts/2020-06-14-dynamic-partition-pruning.md

```diff
@@ -0,0 +1,159 @@
+---
+layout: post
+title:  "Dynamic partition pruning"
+author: Raunaq Morarka, Qubole and Karol Sobczak, Starburst Data
+excerpt_separator: <!--more-->
+---
+
+Improving dynamic filtering to speed up inner-joins on partitioned tables.
+
+<!--more-->
+
+# Introduction
+
+Consider the following query which captures a common pattern of a fact table `store_sales` partitioned by the column 
+`ss_sold_date_sk` joined with a filtered dimension table `date_dim`.
+```
+SELECT COUNT(*) FROM 
+store_sales JOIN date_dim ON store_sales.ss_sold_date_sk = date_dim.d_date_sk
+WHERE d_following_holiday='Y' AND d_year = 2000;
+```
+Without dynamic filtering, Presto will push predicates for the dimension table to the table scan on `date_dim` but 
+it will scan all the data in the fact table since there are no filters on `store_sales` in the query.
+The join operator will end up throwing away most of the probe-side rows as the join criteria is highly selective. 
+The current implementation of [dynamic filtering](https://prestosql.io/blog/2019/06/30/dynamic-filtering.html) improves on
+this by applying dynamic filtering to broadcast joins on tables stored in ORC or Parquet format. 
+But it does not take advantage of the layout of partitioned Hive tables.
+
+With dynamic partition pruning which extends the current implementation of dynamic filtering, every worker node collects 
```

Comment:
I think we should emphasise that dynamic partition pruning does with non-broadcast joins too

Reply:
Change both paragraphs to emphasize the improvements more.